src/pages/CommentVisit.tsx

```diff
@@ -0,0 +1,128 @@
+import 'react-datepicker/dist/react-datepicker.css';
+import { useState } from 'react';
+import { usersVisits } from '../utils/data';
+import { Box, Button, Flex, FormLabel, Select } from '@chakra-ui/react';
+import { FlexForLabelSxStyle, labelSxStyle, mainBoxSxStyle } from './RegisterVisit';
+import SingleSelect from '../components/SingleSelect';
+import { UserVisit, Visit } from '../utils/types';
+import SingleSelectForCombinedArrays from '../components/SingleSelectForCombinedArrays';
+import SingleTextArea from '../components/SingleTextArea';
+
+export default function CommentVisit() {
+  const [databaseUsersVisits, setDatabaseUsersVisits] = useState<UserVisit[]>(usersVisits);
+  const [selectedUser, setSelectedUser] = useState('');
+  const [selectedVisitForUser, setSelectedVisitForUser] = useState<Visit[]>();
+  const [visitArray, setVisitArray] = useState<string[]>();
+  const [textForComment, setTextForComment] = useState('');
+
+  const handleChangeSelectedUser = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    if (e.target.value.length === 0) {
+      setSelectedVisitForUser([]);
+      return;
+    } else {
+      setSelectedUser(e.target.value);
+      const newArrayUser = usersVisits.filter((el) => el.name === e.target.value);
+      setSelectedVisitForUser(newArrayUser[0].visits);
+      if (newArrayUser[0].visits.length === 0) {
+        return;
+      } else {
+        setVisitArray([newArrayUser[0].visits[0].doctor, newArrayUser[0].visits[0].id.toString()]);
+      }
+    }
+  };
+
+  const handleChangeSelectedVisitForUser = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    const newArrayVisit = e.target.value.split(' ');
+    if (newArrayVisit.length === 0 || e.target.value.length === 0) return;
+    setVisitArray(newArrayVisit);
+  };
+
+  const handleTextAreaChange = (e: React.ChangeEventHandler<HTMLTextAreaElement>) => {
+    // jak to otypowac?
+    setTextForComment(e.currentTarget.value);
+  };
+  const handleSubmitComment = (e: React.MouseEvent<HTMLElement>) => {
+    e.preventDefault();
+
+    if (textForComment === '' || visitArray === undefined || selectedVisitForUser === undefined)
+      return;
+
+    usersVisits.map((el) => {
+      if (el.name === selectedUser) {
+        return el.visits.map((el) => {
+          if (el.id === Number(visitArray[1]) && el.doctor === visitArray[0]) {
+            el.comment = textForComment;
+            //ten zapis nic nie zmienial, jak to lepiej zapisac?
+            return { ...el, comment: textForComment };
+          }
+        });
+      }
+    });
+
+    setSelectedUser('');
+    setSelectedVisitForUser([]);
+    setVisitArray([]);
+    setTextForComment('');
+  };
+  return (
+    <>
+      <Box sx={mainBoxSxStyle}>
+        <form>
+          <SingleSelect
+            text={'Select user:'}
+            textPlaceholder={'list of users'}
+            arrayProps={databaseUsersVisits}
+            //?? jak mu pomoc w ts?
+            onChangeProp={handleChangeSelectedUser}
```

Comment:
to tam raczej nie typeGuarda napisać tylko ten typ rozpisać na dwa przypadki jak dostaję tablice UserVisit[] to zawsze będzie React.ChangeEvent<HTMLSelectElement>, a jak Doctor[] to ten drugi typ. Jak coś to pisz to Ci to rozpiszę

Reply:
on ma chyba tutaj glownie problem:

  const handleTextAreaChange = (e: React.ChangeEventHandler<HTMLTextAreaElement>) => {
    // jak to otypowac?
    setTextForComment(e.currentTarget.value);
  };

Reply:
w tym komponencie który jest skomentowany to ogólnie jak masz onChange w środku to tam zawsze będzie changeEvent więc nie wiem czemu rozbiłeś na mouse i change.
Z nowym typem co wyżej wkleiłem currentTarget już działa